cardano-cli/src/Cardano/CLI/EraBased/Run/CreateTestnetData.hs

```diff
@@ -306,18 +307,61 @@ runGenesisCreateTestNetDataCmd Cmd.GenesisCreateTestNetDataCmdArgs
       delegAddrs stuffedUtxoAddrs shelleyGenesis
 
   -- Write genesis.json file to output
-  liftIO $ LBS.writeFile (outputDir </> "genesis.json") $ Aeson.encode shelleyGenesis'
+  liftIO $ LBS.writeFile (outputDir </> "conway-genesis.json") $ Aeson.encode conwayGenesis'
+  liftIO $ LBS.writeFile (outputDir </> "shelley-genesis.json") $ Aeson.encode shelleyGenesis'
+  liftIO $ LBS.writeFile (outputDir </> "alonzo-genesis.json") $ Aeson.encode alonzoGenesis
   where
     genesisDir = outputDir </> "genesis-keys"
     delegateDir = outputDir </> "delegate-keys"
     drepsDir = outputDir </> "drep-keys"
     utxoKeysDir = outputDir </> "utxo-keys"
     poolsDir = outputDir </> "pools-keys"
     stakeDelegatorsDir = outputDir </> "stake-delegators"
-    numStakeDelegators = case stakeDelegators of OnDisk n -> n; Transient n -> n
     mkDelegationMapEntry :: Delegation -> (L.KeyHash L.Staking L.StandardCrypto, L.PoolParams L.StandardCrypto)
     mkDelegationMapEntry d = (dDelegStaking d, dPoolParams d)
 
+    addDRepsToConwayGenesis :: [VerificationKey DRepKey] -> [VerificationKey StakeKey]
+                            -> L.ConwayGenesis L.StandardCrypto -> L.ConwayGenesis L.StandardCrypto
+    addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
+      conwayGenesis { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
+                    , L.cgInitialDReps = initialDReps dRepKeys
+                    }
+
+      where
+      delegs :: [(VerificationKey StakeKey, VerificationKey DRepKey)] -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      delegs = ListMap.fromList . map (bimap verificationKeytoStakeCredential (L.DelegVote . L.DRepCredential . verificationKeyToDRepCredential))
+
+      initialDReps :: [VerificationKey DRepKey] -> ListMap (L.Credential L.DRepRole L.StandardCrypto) (L.DRepState L.StandardCrypto)
+      initialDReps = ListMap.fromList . map (\c -> ( verificationKeyToDRepCredential c
+                                                   , L.DRepState { L.drepExpiry = EpochNo 1000
+                                                                 , L.drepAnchor = SNothing
+                                                                 , L.drepDeposit = L.Coin 100000000
```

Comment:
What happens if this `drepDeposit` is lower than the one defined in genesis? Does this cause any problems when starting the testnet?
Also a small suggestion, would you mind using [`NumericUnderscores`](https://ghc.gitlab.haskell.org/ghc/doc/users_guide/exts/numeric_underscores.html) for large numbers?

Reply:
> What happens if this drepDeposit is lower than the one defined in genesis? Does this cause any problems when starting the testnet?

Ah, you are right, it probably does. I am going to calculate the maximum between `₳100` and that value. I made another PR to `cardano-api` [#485](https://github.com/IntersectMBO/cardano-api/pull/485)